finance.go

```diff
@@ -0,0 +1,46 @@
+package humanize
+
+import (
+	"fmt"
+)
+
+func Finance(f float64) string {
```

Comment:
I wonder if it would make sense to take some optional `Options` which could set things like which $ sign to use and if it should be prepended or appended (or omitted entirely) and could control the # of decimal points.

So you could do things like:

1. `1.83M USD`
2. `€2.34T`
3. `48.9945`

etc. Could also look at the various `locales` used with https://www.php.net/manual/en/function.money-format.php

Reply:
Very good idea, I'll add the Options pattern, the options will be:
- WithSign(s string) // can pass "" to remove the default ($) sign 
- WithAppend() // if no append, Prepend is default -- otherwise could add both options
- WithPrecision(n int)

WDYT?

Reply:
ping @harrisonhjones